core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -164,6 +172,14 @@ object RequestChannel extends Logging {
         m.totalTimeHist.update(totalTime)
       }
 
+      // Record network handler thread usage. This is included towards the request quota for
+      // the user/client. Throttling is only performed when request handler thread usage
+      // is recorded, just before responses are queued for delivery.
+      recordNetworkThreadUsageCallback match {
+        case Some(record) => record(math.max(endTimeNanos - responseDequeueTimeNanos, 0))
```

Comment:
@junrao Thank you, I have reworked the code for recording network thread time as you suggested.  At the moment, SSL/SASL handshake time is included in the time for the first request. Is that reasonable? I can clear the time after authentication if it is confusing.